### PR TITLE
Squash placeholder variables when retrieving the data

### PR DESCRIPTION
**Pull request by stefsmeets**
For #418 , we need to somehow distinguish between different dimensions and variable names so that we do not get name clashes between different IDS. As part of that, I'm looking at the `variables.yaml` file to see how we can make it work a bit better for us. 

One of the things I'm looking at is to simplify how we define the placeholder variables so that we do not have to call `standardize_grid`, and instead do that work invisibly by defining the relations properly.

Everything prefixed `$` in `variables.yaml` is a placeholder that gets replaced when data are loaded. I didn't want to overload `variables.yaml` with a ton of extra keys that make it difficult to update the file later, so I went with this approach.

The goal is to get rid of `standardize_datasets` and `standardize_grid` for the user, and do this automatically.

### Todo

- [x] Update tests
- [x] Update notebooks to demonstrate usage
- [x] Comb through `_rebase.py` and hide `standardize_grid` from public api.
- [x] Update documentation to highlight how `variables.yaml` works
- [x] Follow-up: grep for `variables-*.yaml` and load all variables this way. Can help to add some structure to these files.

### CODE REVIEW

**Comment by stefsmeets** [comment]
@v1kko Could you have a look at this and let me know what you think about this approach?

File: src/duqtools/data/variables.yaml [hunk 1]
```diff
@@ -11,279 +13,288 @@
...
   ids: core_sources
   path: source/0/profiles_1d/*/electrons/particles
-  dims: [time_src, x]
+  dims: [time_src, $rho_tor_norm_src]
   type: IDS-variable
 - name: energy_i_src
   ids: core_sources
   path: source/0/profiles_1d/*/ion/*/energy
-  dims: [time_src, ion_src, x]
+  dims: [time_src, ion_src, $rho_tor_norm]
```
Comment (v1kko) [inline src/duqtools/data/variables.yaml:129]: is this correct, shouldnt it be `rho_tor_norm_src`?

**Review by v1kko — COMMENTED** [review]
Looks great, if this works its a very nice improvement!

File: src/duqtools/data/variables.yaml (hunk 1 above)
Comment (stefsmeets) [reply src/duqtools/data/variables.yaml:129]: Yes, good catch 👍

**Comment by stefsmeets** [comment]
> Looks great, if this works its a very nice improvement!

Cheers, I will keep working on it for a bit then!

**Comment by stefsmeets** [comment]
Alright, I updated the notebooks and it seems to work well for all our use-cases. 🚀

We can rely on `ImasHandle.get_variables()` to load the data, and let it sort out how the data should be squashed.

**Comment by v1kko** [comment]
While working on the merging PR, I realized that `time` is treated differently (no `$` prepended), but I found that it can be treated exactly as 'just another' dimension, should we prepend `time` also with a `$` ?
also see this if statement: https://github.com/duqtools/duqtools/blob/7b31d7c3a6a95da31b5145be8ad976dae1bee5ea/src/duqtools/ids/_merge.py#L43

**Comment by stefsmeets** [comment]
> While working on the merging PR, I realized that `time` is treated differently (no `$` prepended), but I found that it can be treated exactly as 'just another' dimension, should we prepend `time` also with a `$` ? also see this if statement:
> 
> https://github.com/duqtools/duqtools/blob/7b31d7c3a6a95da31b5145be8ad976dae1bee5ea/src/duqtools/ids/_merge.py#L43

Do you mean to combine all time arrays from all IDS? I guess that could work.

**Comment by v1kko** [comment]
Not necessarily combining them from all IDS. (`$time_src` could be different from `$time`). but replacing the dimensions of time-dependent variables (all of them) from:
```
- dims: ['time_src', '$rho_tor_norm']
```
to:
```
- dims:['$time_src', '$rho_tor_norm']
```

**Comment by v1kko** [comment]
Or is this one of the edge cases, where xarray can correctly infer `time` being a coordinate, as it itself has no other coordinate? in that case this is not necessary, I get confused sometimes by these things :confused:

**Comment by stefsmeets** [comment]
I don't know what this would do, because time itself is already a 1D coordinate.

**Pull request merged**